framework/src/Volo.Abp.Imaging.Abstractions/Volo/Abp/Imaging/ImageResizer.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Options;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Threading;
+
+namespace Volo.Abp.Imaging;
+
+public class ImageResizer : IImageResizer, ITransientDependency
+{
+    protected IEnumerable<IImageResizerContributor> ImageResizerContributors { get; }
+    
+    protected ImageResizeOptions ImageResizeOptions { get; }
+    
+    protected ICancellationTokenProvider CancellationTokenProvider { get; }
+    
+    public ImageResizer(
+        IEnumerable<IImageResizerContributor> imageResizerContributors, 
+        IOptions<ImageResizeOptions> imageResizeOptions, 
+        ICancellationTokenProvider cancellationTokenProvider)
+    {
+        ImageResizerContributors = imageResizerContributors;
+        CancellationTokenProvider = cancellationTokenProvider;
+        ImageResizeOptions = imageResizeOptions.Value;
+    }
+    
+    public virtual async Task<ImageResizeResult<Stream>> ResizeAsync(
+        Stream stream, 
```

Comment:
It would be good to add `[NotNull]` and Check.NotNull for the `stream` (also for `bytes` in the next method).
No need to make that everywhere, but here is the entrance point and we should check it.